src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -108,29 +142,16 @@ void PerformanceConfigHipImplicitGemmFwdXdlops::Init(const ProblemDescription& p
     this->total_size = conv_ptrs.size();
     for(int i = 0; i < conv_ptrs.size(); i++)
     {
-        auto argument_ptr = conv_ptrs[i]->MakeArgumentPointer(nullptr,
-                                                              nullptr,
-                                                              nullptr,
-                                                              args.N,
-                                                              args.K,
-                                                              args.C,
-                                                              args.input,
-                                                              args.filter,
-                                                              args.output,
-                                                              args.strides,
-                                                              args.dilation,
-                                                              args.lPadding,
-                                                              args.rPadding,
-                                                              {},
-                                                              {},
-                                                              {});
-        if(conv_ptrs[i]->IsSupportedArgument(argument_ptr.get()))
+        if(args.IsSupportedBy(conv_ptrs[i]))
         {
             this->kernel_id = conv_ptrs[i]->GetTypeString();
             break;
         }
         ++this->index;
     }
+    // TODO(Amber): This logic is broken and different from other
+    // PerformanceConfigHipImplicitsGemm* classes. Other classes keep a vector of
+    // strings called 'valid_kernels'
```

Comment:
It is indeed broken, since `valid_kernels` is used to iterate over performance configs.
I'll take care.

Reply:
Thanks and probably in another PR :)